pallets/asset/src/lib.rs

```diff
@@ -2494,4 +2516,60 @@ impl<T: Config> Module<T> {
             }
         })
     }
+
+    fn base_redeem_from_portfolio(
+        origin: T::Origin,
+        ticker: Ticker,
+        value: Balance,
+        agent_portfolio: PortfolioId,
+    ) -> DispatchResult {
+        // Ensure origin is agent with custody and permissions for default portfolio.
+        let agent = Self::ensure_agent_with_custody_and_perms(origin, ticker)?;
+
+        Self::ensure_granular(&ticker, value)?;
+
+        // Reduce caller's portfolio balance. This makes sure that the caller has enough unlocked tokens.
+        // If `advance_update_balances` fails, `reduce_portfolio_balance` shouldn't modify storage.
+        with_transaction(|| {
+            Portfolio::<T>::reduce_portfolio_balance(&agent_portfolio, &ticker, value)?;
```

Comment:
Security issue: There is no check that `agent_portfolio` belongs to the DID `agent` of the caller `origin`.